Fixes a problem with clock settings of HiFiBerry DAC+ADC PRO

j-schambacher opened the pull request:
This patch fixes a problem of the re-calculation of
i2s-clock and -parameter settings when only the ADC is activated.

Signed-off-by: Joerg Schambacher <joerg@i2audio.com>

j-schambacher committed:
Fixes a problem with clock settings of HiFiBerry DAC+ADC PRO

This patch fixes a problem of the re-calculation of
i2s-clock and -parameter settings when only the ADC is activated.

Signed-off-by: Joerg Schambacher <joerg@i2audio.com>

pelwell commented on sound/soc/bcm/hifiberry_dacplusadcpro.c:
```diff
@@ -414,6 +416,13 @@ static int snd_rpi_hifiberry_dacplusadcpro_hw_params(
+	if (ret)
+		return ret;
+
+	if (snd_rpi_hifiberry_is_dacpro) {
```
You can turn the nested if into an && and lose the braces. Otherwise I'm OK with this patch.

j-schambacher committed:
Fixes a problem with clock settings of HiFiBerry DAC+ADC PRO

This patch fixes a problem of the re-calculation of
i2s-clock and -parameter settings when only the ADC is activated.

Signed-off-by: Joerg Schambacher <joerg@i2audio.com>
